build/ci/goreleaser.yml

```diff
@@ -79,7 +79,20 @@ scoop:
   skip_upload: true
 
 publishers:
-  - name: Upload artifacts to S3
+  - name: Upload artifacts to legacy S3
+    signature: true
+    checksum: true
+    cmd: ./build/package/s3/publish.sh "{{ abs .ArtifactPath }}" "{{ .ArtifactName }}" "{{ .ProjectName }}" "{{ .Version }}" "{{ .Os }}" "{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}"
+    env:
+      - AWS_REGION={{ .Env.AWS_REGION }}
+      - AWS_BUCKET_NAME={{ .Env.AWS_BUCKET_NAME }}
+      - AWS_ACCESS_KEY_ID={{ .Env.AWS_ACCESS_KEY_ID }}
+      - AWS_SECRET_ACCESS_KEY={{ .Env.AWS_SECRET_ACCESS_KEY }}
```

Comment:
@Halama Sem bych potřeboval vygenerovat dočasný credentials. Myslel jsem že předtím zavolám `aws sts get-federation-token` ale lokálně mi to nejde:

![CleanShot 2022-10-12 at 13 14 46](https://user-images.githubusercontent.com/215660/195328696-ca0db5e9-8f10-46ba-841c-e33c30f182da.jpg)

Napadá mě ještě `aws sts assume-role` ale moc nevím jesli by to fungovalo, protože tu roli už tady mám assumnutou, ne? 🤔

Tenhle kód `build/ci/goreleaser.yml` se volá z https://github.com/keboola/keboola-as-code/blob/82e6a5d32ec8cb84a434d5eddc222c9303e359ba/.github/workflows/push.yml#L352-L353


A totéž budu potřebovat pro https://github.com/keboola/keboola-as-code/blob/82e6a5d32ec8cb84a434d5eddc222c9303e359ba/.github/workflows/push.yml#L530-L531 pro který se dělá S3 mount a ten potřebuje credentials strčit do souboru: https://github.com/keboola/keboola-as-code/blob/82e6a5d32ec8cb84a434d5eddc222c9303e359ba/.github/actions/mount-s3/action.yml#L12-L12

Reply:
myslim ze by to mohlo jit zaridit pomoci toho aws-authorize-credentials tomu predradit, mrknu.
Urcite bude do toho mapovani potreba pridat `AWS_SESSION_TOKEN` protoze pujde o docasne credentials.

Reply:
Muzeme to zatim zkusit s tou admin roli co tam je uz autorizovana

Reply:
Ty envy uz tam budou vyexportovane AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN viz. https://github.com/aws-actions/configure-aws-credentials/blob/5e8f93f56c6e1645a5cf01edcf3f9e61343f3fae/index.js#L143

Reply:
Tady je to ale koukam upload do te legacy S3?

Reply:
Jo jo, tohle jo. Zduplikoval jsem to a upload do té nové je od řádku 95. Aha! Hmm, tak tím pádem přejmenuju env vars pro legacy credentials aby nekolidovaly s těma defaultníma (https://github.com/keboola/keboola-as-code/blob/82e6a5d32ec8cb84a434d5eddc222c9303e359ba/.github/workflows/push.yml#L38-L39), na začátek pipeliny dám tu `configure-aws-credentials` akci a tím by mělo být vyřešený. 👍

Reply:
jo nejak tak by to mohlo plus minus fungovat. Kdyztak na to pak muzeme mrknout, ale asi az v patek, zitra nebudu.

Reply:
ok. Díky!

Reply:
@JakubMatejka povedlo se? Můžu tu s něčím pomoct?

Reply:
Já myslím že jo. Resp. asi už nezbude něž to vyzkoušet a udělat release abysme ověřili že to funguje. 🙂

Reply:
jo jasne tohle jede az pri release. Takhle od oka to vypada ok, takze jak pises overit tim releasem.

Reply:
Ok, tak jestli nemáte připomínky ještě k tý struktuře nebo tak něčemu, tak mi to prosím tě approvni a půjdu to vyzkoušet.